master/internal/api_project_intg_test.go

```diff
@@ -525,18 +525,18 @@ func TestConcurrentProjectKeyGenerationAttempts(t *testing.T) {
 	api, _, ctx := setupAPITest(t, nil)
 	wresp, werr := api.PostWorkspace(ctx, &apiv1.PostWorkspaceRequest{Name: uuid.New().String()})
 	require.NoError(t, werr)
-	for x := 0; x < 20; x++ {
-		errgrp := errgroupx.WithContext(ctx)
-		for i := 0; i < 20; i++ {
-			projectName := "test-project" + uuid.New().String()
-			errgrp.Go(func(context.Context) error {
-				_, err := api.PostProject(ctx, &apiv1.PostProjectRequest{
-					Name: projectName, WorkspaceId: wresp.Workspace.Id,
-				})
-				require.NoError(t, err)
-				return err
+	numRequests := 5
+	errgrp := errgroupx.WithContext(ctx)
+	for i := 0; i < numRequests; i++ {
```

Comment:
400 vs 5 is a huge difference...I doubt there will be so many concurrent keygens at once, but was there a requirement to handle any specific number of concurrent keygen attempts?

Reply:
Originally, I had wrote this to be a stress test to ensure that we could handle the case when we start to exhaust the set of project keys for a given project name pattern, but realistically I don't imagine we'll have folks creating 20 projects concurrently 20 times. It being able to support the creation of 5 project synchronously without issue is sufficient for the cadence we expect projects to be created.